Wikipedia/Code/WMFAppViewController.m

```diff
@@ -207,14 +220,74 @@ - (void)pauseApp {
     [self performHousekeepingIfNecessary];
 }
 
+- (void)handleSelectionOfShortcutItemIfNecessaryThen:(void (^)(BOOL shortcutWasHandled))completion {
+    UIApplicationShortcutItem* shortcutItemSelectedAtLaunch = self.shortcutItemSelectedAtLaunch;
+    if (shortcutItemSelectedAtLaunch) {
+        // Main UI may *not* have been loaded if the user...
+        //      - minimized the app when viewing a "Welcome" screen
+        //      - then selected a 3d touch app icon shortcut menu item
```

Comment:
I guess what I am saying is that should we shouldn't try to build the UI if it doesn't exist - that means we haven't completed the on boarding. So is the UI doesnt exist - that should be a logic error. So maybe what we should have is an assertion here to guarantee the UI has been loaded?